File: DbNetSuiteCore/Models/DbNetGridResponse.cs
using System.Collections.Generic;

namespace DbNetSuiteCore.Models
{
    public class DbNetGridResponse
    {
        private List<GridColumn> columns;
        public string Toolbar { get; set; }
        public object Data { get; set; }
        public int TotalPages { get; set; }
        public long TotalRows { get; set; }
        public int CurrentPage { get; set; }
        public List<GridColumn> Columns
        {
            get { return columns; }
            set { columns = value; columns.ForEach(c => c.EncodeClientProperties()); }
        }
        public Dictionary<string, object> Record { get; set; }
        public List<SearchParameter> SearchParams { get; set; }
        public bool Error { get; set; } = false;
        public string Message { get; set; }
    }
}

File: DbNetSuiteCore/Resources/TypeScript/Interfaces/DbNetGridResponse.ts
interface DbNetGridResponse
{
    toolbar: string;
    data: string;
    currentPage: number;
    totalPages: number;
    totalRows: number;
    columns?: GridColumn[];
    record?: Dictionary<string>;
    searchParams?: Array<SearchParam>;
    message?: string;
    error?: boolean;
}

